### PR TITLE
make Reactxp.fs compiled

### PR DESCRIPTION
**Pull request by humhei**
```fsharp   
 <Compile Include="React.fs" />
    <Compile Include="Synctasks.fs" />
    <Compile Include="Subscribableevent.fs" />
    <Compile Include="Reactxp.fs" />
```

### CODE REVIEW

**Comment by humhei** [comment]
I have also published the compiled files in https://github.com/humhei/ReactXPSample

**Comment by alfonsogarciacaro** [comment]
Awesome! :clap: 🥇 👏

File: src/read.fs [hunk 1]
```diff
@@ -8,6 +8,11 @@ open TypeScript.Ts
 open System.Collections.Generic
 open System
 open ts2fable.Naming
+open System.Collections.Generic
+open ts2fable.Syntax
+open System.Collections.Generic
+open Fable
+open System.Collections.Generic
```
Comment (ctaggart) [inline src/read.fs]: looks like duplicate

File: src/TypeScript.fs [hunk 2]
```diff
@@ -2642,7 +2642,9 @@ module Ts =
     type [<AllowNullLiteral>] Statement =
         inherit Node
         abstract _statementBrand: obj option with get, set
-
+        //added for ts2fable
+        abstract nextContainer: Statement option
+        abstract symbol: Statement option
```
Comment (ctaggart) [inline src/TypeScript.fs]: Is there a way to do this without modifying the generated files?

File: src/transform.fs [hunk 3]
```diff
@@ -10,6 +10,10 @@ open System
 open ts2fable.Naming
 open System.Collections
 open System.Collections.Generic
+open ts2fable.Keywords
+open Fable.AST.Babel
+open ts2fable.Read
+open System.Collections.Generic
```
Comment (ctaggart) [inline src/transform.fs]: duplicate

**Comment by ctaggart** [comment]
This is great! I love seeing the additional tests once again. I would like to avoid modifying the generated `Typescript.fs` file manually if possible.

File: src/read.fs (hunk 1 above)
Comment (humhei) [reply src/read.fs]: This is so stupid
The vscode lonide autocomplete feature regularly automatically add it

File: src/TypeScript.fs (hunk 2 above)
Comment (humhei) [reply src/TypeScript.fs]: Remove nextContainer & symbol
This is useless

File: src/TypeScript.fs [hunk 4]
```diff
@@ -3474,7 +3473,8 @@ module Ts =
         abstract getLineStarts: unit -> ResizeArray<float>
         abstract getPositionOfLineAndCharacter: line: float * character: float -> float
         abstract update: newText: string * textChangeRange: TextChangeRange -> SourceFile
-
+        // added for ts2fable
+        abstract resolvedModules: System.Collections.Generic.IDictionary<string,ResolvedModule option> option
```
[the remark this answers is not in the document]
Comment (humhei) [reply src/TypeScript.fs]: I still think it is impossible to get **reslovedModules** by normal way
https://github.com/Microsoft/TypeScript/issues/17546

**Comment by ctaggart** [comment]
@humhei This ready to be merged?

**Comment by humhei** [comment]
Yes.
I am a new one to git
Made a lot of mistakes 
But wo have to go ahead

**Comment by alfonsogarciacaro** [comment]
As @ctaggart is busy now and I'm focused on Fable 2 development, I'm totally OK with giving you write access. Unless Cameron prefers to keep the rights to review and merge the PRs, what do you think @ctaggart?

BTW, after merging #163 it seems there are some conflicts.

**Comment by alfonsogarciacaro** [comment]
Hi @humhei! I sent you an invitation to join the `ts2fable` team so you get write access to this repo. Tell me if you have problems to accept it.

Please send PRs for new changes instead of committing directly to `master` branch, so it's easier to keep track of new features. And please wait a couple of days for me or @ctaggart to review before merging. If we are busy with other things and don't reply, you can merge :wink:

You can create your own branches in this repo if that's more convenient than working on your fork :smile: Just ask me if you have any question, thanks a lot for your help and welcome to the team! :tada: :clap:

**Comment by ctaggart** [comment]
Welcome to the team @humhei! Thanks for picking up where I left off.

**Comment by humhei** [comment]
## Rework:

| PR  | Status |
| ------------- | ------------- |
| #164   |   |

**Comment by humhei** [comment]
@ctaggart @alfonsogarciacaro 
I think i will speed Repo merging up
If you have something want to discuss more.
Please @humhei or add your `test` at `fsfiletest.fs` or add a new issue
😄 
What do you think about this

**Comment by ctaggart** [comment]
Hi @humhei, thanks for splitting this up into multiple issues. It looks like all of them have been merged. As far as I can tell, this is working.

https://github.com/fable-compiler/ts2fable-exports/blob/dev/ReactXP.fs
https://microsoft.github.io/reactxp/

Can this be closed?

**Comment by humhei** [comment]
It should be below codes
A lot work have not been worked out
https://github.com/ts2fable-imports/ReactXPSample/blob/master/src/Imports/ReactXP.fs

**Pull request closed without merging**